plugins/obs-filters/color-correction-filter.c

```diff
@@ -401,7 +386,7 @@ static void color_correction_filter_defaults(obs_data_t *settings)
 	obs_data_set_default_double(settings, SETTING_SATURATION, 0.0);
 	obs_data_set_default_double(settings, SETTING_HUESHIFT, 0.0);
 	obs_data_set_default_double(settings, SETTING_OPACITY, 100.0);
-	obs_data_set_default_int(settings, SETTING_COLOR, 0xFFFFFF);
+	obs_data_set_default_int(settings, SETTING_COLOR, 0x00000000);
```

Comment:
Two concerns:

1. You're changing the default value for the color setting, which could affect people upgrading, wouldn't it?
1. The "Opacity" value still seems to exist, which is sort of a duplicated functionality isn't it?

Reply:
This is a different opacity. The alpha value of "Color" controls how of the image is obscured by the RGB part.

I changed the default to be less confusing. It's currently "transparent white," which does nothing since 100% original image + 0% white is 100% original image. 0% white or 0% black makes no difference, but I wanted to make it clear the white didn't mean anything.

Reply:
Hm, I understand now. For the sake of completeness, did you test upgrading from a color correction filter that had a color set to see if the before/after of using one version to using the new version match?

Reply:
The value carries over, but It's not going to match because I tweaked the math to do a regular alpha blend.

(Not completely regular since it's not linear-correct, but that's for another day.)

Before RGB: (color.rgb * color.a) + (image.rgb * color.rgb)
Before A: image.a

After RGB: (color.rgb * color.a) + (image.rgb * (1 - color.a))
After A: color.a + (image.a * (1 - color.a))

Reply:
The important thing is mostly that it doesn't do something significantly different. I'll check on my end as well, but as long as it's not significantly different then it's fine. I mostly just want to ensure that anyone upgrading isn't noticeably affected

Reply:
It can be pretty different, but the old formula didn't make much sense.

Reply:
I used colour overlay a year ago to make a black and white capture mimic a game boy LCD screen and can confirm the controls made no logical sense to me. I could get a desirable result but it was super user unfriendly.

Reply:
I was planning to deprecate this filter entirely for a new color correction filter after the SRGB PR went though. I can change the math back, and we can wait until then to do the breaking change. Having a color picker with alpha would still be necessary though.

Reply:
I'm fine with whatever as long as people upgrading don't experience any change in behavior. I'd say just leave the color filter as it was, and when the sRGB stuff goes through, make a new version and deprecate the old one (while still leaving the old one's prior functionality/code intact so people upgrading don't have a sudden change). Do a `git grep OBS_SOURCE_CAP_OBSOLETE` to see how we normally change the functionality.

Although it results in old code lingering, doing it that way would allow you to also make a lot more changes that you think make sense, because personally I'd trust your judgment with color-related stuff.

Reply:
Updated the PR to maintain existing behavior.

Reply:
Actually, there was still one minor regression, but it's fixed now.